plugins/google-sheets/src/pages/MapSheetFields.tsx

```diff
@@ -56,7 +56,8 @@ const inferFieldType = (cellValue: CellValue): CollectionFieldType => {
 
         // If the cell value contains a newline, it's probably a formatted text field
         if (cellValueLowered.includes("\n")) return "formattedText"
-        if (/^\d{1,2}-\d{1,2}-\d{4}/.test(cellValueLowered)) return "date"
+        const maybeDate = new Date(cellValueLowered)
+        if (!isNaN(maybeDate.getTime())) return "date"
```

Comment:
Nit: Better version of `isNaN`, but make sure it behaves how we want:

```suggestion
        if (!Number.isNaN(maybeDate.getTime())) return "date"
```